src/EditorFeatures/Core/Classification/CopyPasteAndPrintingClassificationBufferTaggerProvider.Tagger.cs

```diff
@@ -150,18 +159,50 @@ public IEnumerable<ITagSpan<IClassificationTag>> GetAllTags(NormalizedSnapshotSp
                 }
             }
 
-            return SegmentedListPool.ComputeList(
+            return SegmentedListPool<ITagSpan<IClassificationTag>>.ComputeList(
                 static (args, tags) => args.cachedTags?.AddIntersectingTagSpans(args.spans, tags),
-                (cachedTags, spans),
-                _: (ITagSpan<IClassificationTag>?)null);
-        }
+                (cachedTags, spans));
 
-        private Task ProduceTagsAsync(
-            TaggerContext<IClassificationTag> context, DocumentSnapshotSpan snapshotSpan,
-            IClassificationService classificationService, ClassificationOptions options, ClassificationType type, CancellationToken cancellationToken)
-        {
-            return ClassificationUtilities.ProduceTagsAsync(
-                context, snapshotSpan, classificationService, _owner._typeMap, options, type, cancellationToken);
+            async ValueTask AddSyntacticSpansAsync(NormalizedSnapshotSpanCollection spans, SegmentedList<ITagSpan<IClassificationTag>> result, bool _)
+            {
+                Contract.ThrowIfTrue(spans.Count != 1, "We should only be asking for a single span when getting the syntactic classifications");
+
+                await AddSpansAsync(spans, result,
+                    span => classificationService.AddSyntacticClassificationsAsync(document, span, tempClassifiedSpans, cancellationToken)).ConfigureAwait(false);
+            }
+
+            async ValueTask AddSemanticSpansAsync(NormalizedSnapshotSpanCollection spans, SegmentedList<ITagSpan<IClassificationTag>> result, bool _)
+            {
+                Contract.ThrowIfTrue(spans.Count != 1, "We should only be asking for a single span when getting the semantic classifications");
+
+                await AddSpansAsync(spans, result,
+                    span => classificationService.AddSemanticClassificationsAsync(document, span, options, tempClassifiedSpans, cancellationToken)).ConfigureAwait(false);
+            }
+
+            async ValueTask AddEmbeddedSpansAsync(NormalizedSnapshotSpanCollection stringLiteralSpans, SegmentedList<ITagSpan<IClassificationTag>> result, bool _)
```

Comment:
💡 It seems strange to have the `TArg` parameter here but not actually use it to make the method `static`

Reply:
enh.  the amount of data needed to pass in in all these cases was a lot (something like 6 parameters).  My pref is to keep things simple.  The mainline classification case uses statics for efficiency.  But this is for copy/paste, which is a user invoked operation, so 3 extra allocs is fine :)